### ISSUE TITLE
Add Neural Network functions for TensorFlow frontend

### ISSUE THREAD

**Issue by sreegandh804** [unifyai/ivy#1548]
Add [Neural Network functions](https://www.tensorflow.org/api_docs/python/tf/nn#functions) for TensorFlow frontend:
- [ ] atrous_conv2d
- [ ] atrous_conv2d_transpose
- [ ] avg_pool1d
- [ ] avg_pool2d
- [ ] batch_normalization
- [ ] collapse_repeated
- [ ] compute_average_loss
- [ ] conv1d
- [ ] conv1d_transpose
- [ ] conv2d_transpose
- [ ] conv3d_transpose
- [ ] conv_transpose
- [ ] convolution
- [ ] crelu
- [ ] ctc_unique_labels
- [ ] depthwise_conv2d
- [ ] depthwise_conv2d_backprop_filter
- [ ] depthwise_conv2d_backprop_input
- [ ] dropout
- [ ] embedding_lookup
- [ ] embedding_lookup_sparse
- [ ] erosion2d
- [ ] gelu
- [ ] l2_normalize
- [ ] local_response_normalization
- [ ] log_poisson_loss
- [ ] max_pool1d
- [ ] max_pool2d
- [ ] moments
- [ ] nce_loss
- [ ] normalize_moments
- [ ] pool
- [ ] safe_embedding_lookup_sparse
- [ ] sampled_softmax_loss
- [ ] scale_regularization_loss
- [ ] separable_conv2d
- [ ] sigmoid_cross_entropy_with_logits
- [ ] silu
- [ ] sufficient_statistics
- [ ] swish
- [ ] weighted_cross_entropy_with_logits
- [ ] weighted_moments
- [ ] with_space_to_batch
- [ ] zero_fraction

**Comment by VikramxD** [comment]
- [ ] #2424

**Comment by ToobaJamal** [comment]
#2720

**Comment by ToobaJamal** [comment]
#2721

**Comment by sherry30** [comment]
- [ ] #3389

**Comment by sherry30** [comment]
- [ ] #3390

**Comment by nj19257** [comment]
avg_pool2d  #3845

**Comment by Change-m** [comment]
avg_pool1d #4393

**Comment by Change-m** [comment]
collapse_repeated #4396

**Comment by Change-m** [comment]
compute_average_loss #4406

**Comment by Change-m** [comment]
conv_transpose #4420

**Comment by Change-m** [comment]
convolution #4426

**Comment by Change-m** [comment]
depthwise_conv2d #4469

**Comment by Change-m** [comment]
depthwise_conv2d_backprop_filter #4471

**Comment by Change-m** [comment]
depthwise_conv2d_backprop_input #4473

**Comment by Change-m** [comment]
dropout #4475

**Comment by Aditya-Komaravolu** [comment]
max_pool1d #4477

**Comment by Change-m** [comment]
ctc_unique_labels #4478

**Comment by rannynayyar67** [comment]
#4810

**Comment by Chainke52** [comment]
#4994

**Comment by hmahmood24** [comment]
- [ ] #5084

**Comment by hmahmood24** [comment]
- [ ] #5107

**Comment by hmahmood24** [comment]
- [ ] #5149

**Comment by yis11178** [comment]
- [ ] #Issue_5309

**Comment by Mina-88** [comment]
- [ ] #5491

**Comment by MutugiD** [comment]
conv_transpose #6353

**Comment by mohamedehab0999** [comment]
- [ ]  #6704

**Comment by hamehrabi** [comment]
#6736

**Comment by vishalbalaji-v** [comment]
Hi. What is the difference between the `swish` here and the `swish` in [activation functions](https://github.com/unifyai/ivy/issues/1569)

**Comment by saif-ashraf99** [comment]
#7302

**Comment by vishalbalaji-v** [comment]
Moments - #7349

**Comment by aylinaydincs** [comment]
l2_normalize - #7373

**Comment by guptaru1** [comment]
conv2d #4176

**Comment by nirajkulkarni26** [comment]
max_pool1d#7441

**Comment by nirajkulkarni26** [comment]
#7441

**Comment by anurag629** [comment]
- [ ] #7538

**Comment by ra9hur** [comment]
#7598

**Comment by ra9hur** [comment]
depthwise_conv2d #7598

**Comment by Hypatchia** [comment]
avg_pool1d#7765

**Comment by abhayKashyap03** [comment]
[ ] gelu #7879

**Comment by ashwaqhussain12** [comment]
avg_pool2d #6796

**Comment by ashwaqhussain12** [comment]
[] #6796

**Comment by ashwaqhussain12** [comment]
- [ ] #6796

**Comment by SaiAakash** [comment]
avg_pool1d #10628

**Comment by ajithpious** [comment]
- [ ] #11775

**Comment by patpanda94** [comment]
"- [ ] #13207"

**Comment by omarahmed19992** [comment]
- [ ] #13818

**Comment by EkaterinaPolishchuk** [comment]
- [ ] #13839

**Comment by JoaoPicolo** [comment]
- [ ] #13961

**Comment by JoaoPicolo** [comment]
- [ ] #14015

**Comment by kaushal07wick** [comment]
hey @ivy-seed the `moments` function of the tensorflow.nn should be in `ivy/ivy/functional/frontends/torch/nn/functional/norms.py`
right!

**Comment by jonotassia** [comment]
- [ ] #14935

**Comment by divz33** [comment]
embedding_lookup_sparse #15285

**Comment by bilalboukerma** [comment]
Hello, Unify ai. 
the status of the task pool. Is it still available? If so, I want to claim the task pool with my issue pool #15398. If not, please let me know so I can select another task.

**Comment by mahmodKhaled** [comment]
pool #15879

**Comment by RohitPawar2406** [comment]
- [ ] #15930

**Comment by LateefAkinola** [comment]
- [ ] #15942

**Comment by usayrimrehman** [comment]
- #15974

**Comment by RohitPawar2406** [comment]
- [ ] #16080

**Comment by Klb21** [comment]
- [ ] #16402

**Comment by Klb21** [comment]
- [ ] #16444

**Comment by enjay-k** [comment]
- [ ] #16711

**Comment by rullo16** [comment]
- [ ] #16743

**Comment by alrizkipascar** [comment]
- [ ] [normalize moments #17188](https://github.com/unifyai/ivy/issues/17188)

**Comment by alrizkipascar** [comment]
* [ ]  [ctc_unique_labels #17626](https://github.com/unifyai/ivy/issues/17626)

**Comment by eekamal** [comment]
- [ ] #17671

**Comment by samthakur587** [comment]
#18304

**Comment by mosiiiiiii** [comment]
- [ ] #19219

**Comment by RasulAlakbarli** [comment]
#19558

**Comment by usayrimrehman** [comment]
#21016

**Comment by berkantkaplan** [comment]
#21592

**Comment by usayrimrehman** [comment]
#21872

**Comment by rahulkusuma1999** [comment]
- [ ] #22276

**Comment by dettmerramon** [comment]
- [ ] #1548

**Comment by DoudGaya** [comment]
#22669

**Comment by ANIRUDH-333** [comment]
Can I try ```compute_average_loss```? I am new to open source contributions, so I don't know much about how to take up a task. Can I work on that function?

**Comment by kahina45678** [comment]
#22783

**Comment by anmol-narang77** [comment]
#23170

**Comment by usayrimrehman** [comment]
#23172

**Comment by bhavya092** [comment]
- [ ]  #23317

**Comment by Stag51** [comment]
- [ ] #23354

**Comment by aps19** [comment]
#23727

**Comment by Dofften** [comment]
- [ ] #23741

**Comment by abdul-rehman18** [comment]
#25773

**Comment by Karanveer2202** [comment]
#26189

**Comment by romailafzal** [comment]
#26246

**Comment by romailafzal** [comment]
- [ ] #26246

**Comment by romailafzal** [comment]
> * [ ]  [depthwise_conv2d_backprop_input #26246](https://github.com/unifyai/ivy/issues/26246)

is this correct way to open an issue?

**Comment by TalhaKhalil** [comment]
#23741
